### ISSUE TITLE
Pax-Exam should fail test when a test-provisioned bundle fails to resolve/start [PAXEXAM-433]

### ISSUE THREAD

**Issue by ops4j-issues** [ops4j/org.ops4j.pax.exam2#559]
**[Peter Gardfjall](https://ops4j1.jira.com/secure/ViewProfile.jspa?accountId=557058:3743539f-aa70-417c-a960-61e0fe87108a)** created **[PAXEXAM-433](https://ops4j1.jira.com/browse/PAXEXAM-433)** 

Pax-Exam (2.5.0) doesn't fail/warn me when a test-provisioned bundle fails to start/resolve (missing constraint).<br /> In my case, I end up with a ClassNotFoundException when my test attempts to use a class from the bundle.

I would have expected Pax-Exam to fail when a provisioned bundle fails to resolve.<br /> That would shorten the troubleshooting time quite drastically.

The attached Maven project contains a test case that illustrates the problem.<br /> The test code makes use of Apache Commons DBCP but forgets to provision Apache Commons Pool (which is a dependency of the former) to the OSGi test framework.<br /> This results in a ClassNotFoundException, whereas it would have been more reasonable to have it fail the test with an error saying that some provisioned bundles (Apache Commons DBCP in this case) failed to resolve.

In summary: Pax-Exam should fail hard when a test-provisioned bundle fails to resolve.

---

**Affects:** 2.5.0
**Fixed in:** 3.0.0
**Attachments:**
* [bug.zip](https://ops4j1.jira.com/secure/attachment/11331/bug.zip)

Votes: 0, Watches: 2

---
**Referenced issues**

**depends on:**
* ops4j/org.ops4j.pax.swissbox#77 - Query bundle state in remote framework [PAXSB-64]
* #595 - Upgrade to Pax Swissbox 1.6.0 [PAXEXAM-470]

**is duplicated by:**
* #276 - tests should fail on missing wiring requirement [PAXEXAM-499]
